logstash-core/src/main/java/org/logstash/config/ir/PipelineIR.java

```diff
@@ -51,7 +51,7 @@ public PipelineIR(Graph inputSection, Graph filterSection, Graph outputSection,
 
         this.graph.validate();
 
-        if (this.getOriginalSource() != null && this.getOriginalSource().matches("^\\S+$")) {
+        if (this.getOriginalSource() != null && !this.getOriginalSource().matches("^\\s+$")) {
```

Comment:
Ah okay, I misunderstood the title/description of the PR. I thought it meant if _any_ spaces were present. Thanks for clarifying!